pkg/ddl/index.go

```diff
@@ -1065,6 +1339,9 @@ func runReorgJobAndHandleErr(
 			func() {
 				addIndexErr = dbterror.ErrCancelledDDLJob.GenWithStack("add table `%v` index `%v` panic", tbl.Meta().Name, allIndexInfos[0].Name)
 			}, false)
+		if job.Type == model.ActionAddVectorIndex {
+			return w.checkVectorIndexProcess(tbl, reorgInfo)
```

Comment:
This can be moved to the place where calling `doReorgWorkForCreateIndexOnTiFlash`. `reorgInfo` and `isReorgRunnable` is not necessary.

Reply:
RowCount also needs to be synchronized, so the unified `runReorgJobAndHandleErr` method is used directly.
Why isn't isReorgRunnable needed? You still need this check if you want to Cancel or something like that.

Reply:
> RowCount also needs to be synchronized

`job.SetRowCount()` should be enough.

> Why isn't isReorgRunnable needed? You still need this check if you want to Cancel

OK. But we can still use `worker.ddlCtx.isReorgRunnable()` outside.

Reply:
Done.